src/cinder/linux/GstPlayer.cpp

```diff
@@ -923,14 +923,13 @@ bool GstPlayer::setPipelineState( GstState targetState )
 	GstStateChangeReturn stateChangeResult = gst_element_set_state( mGstData.pipeline, mGstData.targetState );
 	g_print( "Pipeline state about to change from : %s to %s\n", gst_element_state_get_name( current ), gst_element_state_get_name ( targetState ) );
 
-	bool success = checkStateChange( stateChangeResult );
 	if( ! sEnableAsyncStateChange && stateChangeResult == GST_STATE_CHANGE_ASYNC ) {
-		gst_element_get_state( mGstData.pipeline, &current, &pending, GST_CLOCK_TIME_NONE );
-		mGstData.updateState( current );
-		g_print( " Passed blocking state change to : %s with pending : %s\n", gst_element_state_get_name( current ), gst_element_state_get_name ( targetState ) );
-	}
-	
-	return success;
+        g_print( " Blocking until pipeline state changes from : %s to %s\n", gst_element_state_get_name( current ), gst_element_state_get_name ( targetState ) );
+        stateChangeResult = gst_element_get_state( mGstData.pipeline, &current, &pending, GST_CLOCK_TIME_NONE );
+        mGstData.updateState( current );

```

Comment:
Thanks for this Patrick!

With this ordering there is also no need for the manual call to `updateState` here which is nice. Do you want to remove it and and include the change in this PR ? Otherwise I can add it after the merge.

Also, for future reference in terms of code style, tabs vs spaces and so on you can have a look [here](https://github.com/cinder/Cinder/blob/master/CONTRIBUTING.md) .